chef/cookbooks/hyperv/attributes/default.rb

```diff
@@ -2,11 +2,31 @@
 # Windows features required:
 #
 
-default[:features_list][:hyperv] = ["Microsoft-Hyper-V", "Microsoft-Hyper-V-Management-PowerShell"]
-default[:features_list][:management] = ["Remote-Desktop-Services"]
-default[:features_list][:iscsi_target] = ["File-Services", "CoreFileServer"]
+default[:features_list][:hyperv] = {
+  "Microsoft-Hyper-V" => {
+    "restart" => false
+  },
+  "Microsoft-Hyper-V-Management-PowerShell" => {
+    "restart" => true
```

Comment:
@vuntz no, this is not too careful :-) , it's normal. Normally chef-client will be stopped when a reboot is required and continue after boot. I presume this will not be an easy fix, but from the process point of view it makes total sense.